### DISCUSSION TITLE
Analyze Blocking Argument and Timeouts

### DISCUSSION THREAD

**Discussion by nick-ra** [ansys/pyaedt#3760] [Q&A] [answered]
I tried setting the blocking=False argument in the Maxwell3d.analyze function when running a parametric sweep and it didn't seem to have an effect. I wonder, am I misunderstanding the use of the blocking argument? I am using 2023R2 and pyaedt 0.7.

My larger goal is to implement a timeout feature because I have a simulation which is a bit flaky and a few of the design points in the large set of runs seem to hang randomly (but weirdly run fine if I re-run them).

I ran the analysis with the following commands expecting the script to continue and evaluate whether the model is still running but the function wouldn't continue until the analysis finished completely:
```
    timeout = 30
    maxwell3d.analyze(analysis, num_cores=8, num_tasks=8, num_gpu=0, use_auto_settings=False, blocking=False)
    is_running = True
    mytime = 1
    run_complete = False
    while is_running:
        time.sleep(10) # wait 10s
        mytime = mytime + 10 # increment time
        print(mytime, maxwell3d.are_there_simulations_running) # test
        if maxwell3d.are_there_simulations_running == 0: # check if simulations are complete
            run_complete = True
            print("run complete")
            break
        elif mytime > timeout: # check if timeout is reached
            maxwell3d.stop_simulations(False) # stop simulations
            run_complete = False
            break
```

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @nick-ra,

I can not reproduce the issue. With the attached project:

[Maxwell3D.zip](https://github.com/ansys/pyaedt/files/13053328/Maxwell3D.zip)

I opened and then I run this code:

```
from pyaedt import Maxwell3d

aedt_version = '2023.2'

hfss = Maxwell3d(specified_version=aedt_version, designname="1_Magnetic")

hfss.analyze("Setup1", num_cores=6, num_tasks=6, num_gpu=0, use_auto_settings=False, blocking=False)

hfss.release_desktop(False, False)
```

I am using PyAEDT v0.7.1

**Reply by nick-ra** [reply]
Hi, thanks for this example. This works great as is but it doesn't for Parametric variations under Optimetrics. I added a parametric setup to your example and solving that instead, the script doesn't continue until the solution is done. Is that an intended limitation of this method?
[Maxwell3DParametric.zip](https://github.com/ansys/pyaedt/files/13078676/Maxwell3DParametric.zip)

**Reply by Samuelopez-ansys** [reply]
@nick-ra The parametric analysis does not have this feature, only the analyze has it. Please contact your Ansys technical point to submit an enhancement request. PyAEDT can just use what is integrated in AEDT.

If you want to run in sequence some parameters, you can just sweep the parameter with the python script instead of using Optimetrics.

**Reply by nick-ra** [reply]
OK, that helps, thank you for the explanation.
